core/includes/file.inc

```diff
@@ -1343,6 +1343,13 @@ function file_create_filename($basename, $directory) {
  * @see file_usage_list()
  */
 function file_delete($fid) {
+  // Add backwards-compatible parameter handling.
+  global $settings;
+  if ($settings['backdrop_drupal_compatibility'] && is_a($fid, 'File')) {
+    // @todo once https://github.com/backdrop/backdrop-issues/issues/4220 is
+    // resolved, notify developers of this change.
```

Comment:
I'd suggest keeping this comment out of core and just putting a note in https://github.com/backdrop/backdrop-issues/issues/4220 that `file_delete()` needs updating if/when it gets committed.

```suggestion
```

Reply:
I personally don't mind leaving that `@todo` here. There's chances that https://github.com/backdrop/backdrop-issues/issues/4220 gets fixed and closed without this ever being addressed.